src/wlapplication.cc

```diff
@@ -1074,7 +1074,8 @@ void WLApplication::warp_mouse(const Vector2i position) {
 		SDL_Window* sdl_window = g_gr->get_sdlwindow();
 		if (sdl_window != nullptr) {
 			if (!mouse_locked_) {
-				SDL_PumpEvents();
+				// Fix for #5655, TODO(klaus.halfmmann): shall we do this for macOS only?
+				NoteThreadSafeFunction::instantiate([]() { SDL_PumpEvents(); }, true);
 				SDL_FlushEvent(SDL_MOUSEMOTION);
 				SDL_WarpMouseInWindow(sdl_window, position.x, position.y);
```

Comment:
Please move the other SDL calls into the lambda as well. And remove the TODO – this is not mac-specific. Then this can be merged. I'll take care of backporting to 1.1 afterwards.

Reply:
Thanks for locating the bug and fix, I am eager to test on macOS.

Reply:
Sorry I can only continue working / arguing with @Noordfrees next year. 
For me this fix is complete so please test it. Wrapping other SDL functions will only slow down the Game.
We need to merge this into 1.1 and then get the brew formular going.

@hillierdani  do you have time/resources to support us?